bin/async-core.js

```diff
@@ -0,0 +1,115 @@
+"use strict"
+
+const o = require("../ospec")
+
+const glob = require("glob")
+const os = require("os")
+const path = require("path")
+const {Worker} = require("worker_threads")
+
+module.exports = async function asyncCore({globList, ignore, parallel, dependencies, cwd}) {
+	if (dependencies) await Promise.all(
+		dependencies.filter((dep) => dep != null).map(
+			(module) => import(require.resolve(module, {paths: [cwd]}))
+		)
+	)
+
+	var cores = parallel ? os.cpus().length : 1
```

Comment:
`s/cores/processes/g` - Let's not tie this strictly to literal cores, as that's not how OS threads are handled. (Operating system threads don't even map one-to-one to hardware threads, and hyperthreading makes hardware threads not even map one-to-one with hardware cores.)

Also, from my own personal experimentation with various tools and such (as I've written quite a bit of parallel code in the past), I've found that a good balanced default for this for 2-4 cores is actually `numCPUs + 1`, not `numCPUs` - FS syscalls can come up in plenty of unexpected places, but also some things do end up CPU-bound. But in general, CPU-bound applications don't suffer too much when only one thread is competing with them for resources, and that thread does eventually get scheduled.